### PR TITLE
Move ToInterfaceName from sdk-vpp for kernel mechanism

### PR DESCRIPTION
**Pull request by glazychev-art**
https://github.com/networkservicemesh/sdk-vpp/issues/254

Signed-off-by: Artem Glazychev <artem.glazychev@xored.com>

### CODE REVIEW

**Comment by denis-tingaikin** [comment]
@pperiyasamy, @edwarnicke Please take a look at this PR

**Review by pperiyasamy — APPROVED** [review]

File: pkg/api/networkservice/mechanisms/kernel/helpers.go [hunk 1]
```diff
@@ -86,6 +86,21 @@ func (m *Mechanism) IsPCIDevice() bool {
...
+	// if we are on the client side, we want to name based on the connection id from the NSE that is Next
+	// if we are not the client, we want to name for the connection of of the client addressing us, which is Prev
+	namingConn := conn.Clone()
+	namingConn.Id = namingConn.GetPrevPathSegment().GetId()
+	if isClient {
+		namingConn.Id = namingConn.GetNextPathSegment().GetId()
+	}
+	return m.GetInterfaceName(namingConn)
+}
+
```
Comment (edwarnicke) [inline pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: While this is precisely what is currently being done in sdk-vpp, it does not give the correct results.

The desired outcome is to use the connection ID of the actual Client in the chain if isClient is false (meaning its being used from a 'server' side of a forwarder) and to use the connection ID of the Endpoint if isClient is true (meaning that its the 'client' side of a forwarder).

Take the example where the Client -> NSMgr -> Forwarder -> NSMgr -> Endpoint
In that example, the forwarder will use the connection ID of the NSMgr and for both values of isClient, and not the actual client or Endpoint.

What is needed here is a bit of a rethink.

One possible things to be rethought:
- Maybe we have the Client or Endpoint set the name as part of the kernel mechanism chain elements from sdk?

I'd love to see other ideas here for to get to the correct result.

Comment (pperiyasamy) [reply pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: @edwarnicke is there a way to get client or endpoint connection object in forwarder ? or do you want endpoint itself set the interface name in mechanism parameter like client does it ?

Comment (glazychev-art) [reply pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: @edwarnicke 
I would suggest to move interface-name generation from `api` to `sdk`. And set the default name for the kernel interface.
Created the draft PR for `sdk`:
https://github.com/networkservicemesh/sdk/pull/1014

Comment (edwarnicke) [reply pkg/api/networkservice/mechanisms/kernel/helpers.go:103]: @glazychev-art that looks like the right direction... I've commented other there.

**Pull request closed without merging**